docs/en/observability/ci-cd-observability.asciidoc

```diff
@@ -427,3 +427,53 @@ The Concourse CI doesn’t report health metrics through OpenTelemetry. However,
 https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/spanmetricsprocessor#span-metrics-processor[OpenTelemetry Collector Span Metrics Processor]
 to derive pipeline execution traces into KPI metrics like throughput and the error rate
 of pipelines.
+
+
+[discrete]
+[[check-service-health-from-cd-pipelines]]
+== Check service health from deployment pipelines
+
+
+Integrating automated service health checks in deployment pipelines is critical to end-to-end deployment automation which is critical to increase the deployment frequency .
+
+Elastic Observability exposes service HTTP health check APIs that that integrate in deployment pipelines to verify the behaviour of newly deployed instances
+and automatically continue the deployments or rollback according to the health statuses.
+
+
+image::images/ci-cd-canary-deployment-pipeline.png[Canary Deployment Pipeline]
+
+
+The primary HTTP API to check the health of a service is compare the transaction error rate with a threshold value.
+The API call requires authentication, we recomend to use an API Token to authenticate.
+The URL is `KIBANA_URL/internal/apm/services`, it is designed to query service metrics.
+In order to check the transaction error rate of the service on the newly deployed instance, we have to pass the following parameters to the API call:
+* `start` and `end`: time interval using the ISO-8601 format
+* `kuery`: used to filter on the service name and for example the new version being deployed to narrow down to the canary instances. Example `service.name:"MY_SERVICE_NAME" AND service.version:"1.2.3"`
+* `environment`: the environment on which the canary instances are deployed. Example: `production`.
+We need a rande of time, this range is defined by the parameters `start` and `end`, 
+these parameters are two dates in ISO-8601 format. Decause we want to query only one service,
+we have to compose a filter in the param `kuery`, we can filter by the service name using the expresion `service.name:MY_SERVICE_NAME`.
+Finally we apply a filter by environment passing the parameter `environment`, 
+to select all environments the value is `ENVIRONMENT_ALL`
+
+The API is subject to changes and a stable API optimized for Continous Delivery use cases will soon be published.
+
+[source,python]
+----
+def check_service_health(service_name, error_rate_threshold, kibana_url, api_token):
+    now = datetime.now()
+    five_minutes_ago = now - timedelta(minutes=5)
+    params = {
+        "start": five_minutes_ago.strftime("%Y-%m-%dT%H:%M:%SZ"),
```

Comment:
@cyrille-leclerc Here is the format pattern of the ISO-8601 date format

Reply:
so it could be like "2021-09-01T13:24:12Z" for an UTC time, correct?

Reply:
Yes, that the format "YYYY-MM-DDThh:mm:ssZ"

https://en.wikipedia.org/wiki/ISO_8601
https://www.iso.org/iso-8601-date-and-time-format.html

Reply:
thanks, closing for the moment.